src/sentry/utils/event_tracker.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from abc import ABC, abstractmethod
+from enum import Enum
+
+
+class EventStageStatus(Enum):
+    START = "start"
+    END = "end"
```

Comment:
Consider making the enum an integer. That would take a lot less memory in redis.

The galaxy brain version would be to create a bitmap. One bit per phase. Each event only needs 12 bits to encode all phases. https://redis.io/docs/latest/develop/data-types/bitmaps/

Reply:
not writing to redis with the logging implementation but i get the idea